src/mongoose_wpool.erl

```diff
@@ -137,8 +184,60 @@ get_pools() ->
 stats(Type, Host, Tag) ->
     wpool:stats(make_pool_name(Type, Host, Tag)).
 
+-spec make_pool_name(type(), host(), tag()) -> atom().
 make_pool_name(Type, Host, Tag) when is_atom(Host) ->
     make_pool_name(Type, atom_to_binary(Host, utf8), Tag);
 make_pool_name(Type, Host, Tag) when is_binary(Host) ->
     binary_to_atom(<<"mongoose_wpool$", (atom_to_binary(Type, utf8))/binary, $$,
                      Host/binary, $$, (atom_to_binary(Tag, utf8))/binary>>, utf8).
+
+start_by_callback(Type, Host, Tag, WpoolOpts, ConnOpts) ->
+    try
+        CallbackModule = make_callback_module_name(Type),
+        CallbackModule:start(Host, Tag, WpoolOpts, ConnOpts)
+    catch E:R ->
+          ?ERROR_MSG("event=wpool_callback_start_error, error=~p, reason=~p, stacktrace=~p",
+                     [E, R, erlang:get_stacktrace()]),
+          {error, pool_not_started}
+    end.
+
+init(Type) ->
+    try
+        CallbackModule = make_callback_module_name(Type),
+        CallbackModule:init()
+    catch E:R ->
+          ?ERROR_MSG("event=wpool_callback_init_error, error=~p, reason=~p, stacktrace=~p",
+                     [E, R, erlang:get_stacktrace()]),
+          {error, pool_not_initialized}
+    end.
+
+stop_by_callback(Type, Host, Tag) ->
+    try
+        CallbackModule = make_callback_module_name(Type),
+        CallbackModule:stop(Host, Tag)
+    catch E:R ->
```

Comment:
Feels like this try/catch dance could be deduplicated

Reply:
I'll try to deduplicate it.